### PR TITLE
fv v2 identifier

### PR DESCRIPTION
**Pull request by sirpy**
# Description
Verify correctly new format of FV identifiers that are tied to the whitelisted address

### CODE REVIEW

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -16,10 +16,20 @@ import { sendTemplateEmail } from '../aws-ses/aws-ses'
...
 
 import createEnrollmentProcessor from './processor/EnrollmentProcessor.js'
-import { verifySignature } from '../utils/eth'
+import { recoverPublickey } from '../utils/eth'
 import { shouldLogVerificaitonError } from './utils/logger'
 import { syncUserEmail } from '../storage/addUserSteps'
-
+import { FV_IDENTIFIER_MSG2 } from '../login/login-middleware'
+
+const verifyFVIdentifier = async (identifier, gdAddress) => {
```
Comment (johnsmith-gooddollar) [inline src/server/verification/verificationAPI.js:24]: this should be moved to src/server/verification/utils

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -69,18 +96,25 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
     '/verify/face/:enrollmentIdentifier',
     passport.authenticate('jwt', { session: false }),
     wrapAsync(async (req, res) => {
-      const { params, log, user } = req
+      const { params, log, user, query } = req
```
Comment (johnsmith-gooddollar) [inline src/server/verification/verificationAPI.js:99]: see suggestions from above

File: src/server/verification/verificationAPI.js [hunk 3]
```diff
@@ -171,14 +206,18 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
       user.chainId = chainId || conf.defaultWhitelistChainId
 
       try {
+        // for v2 identifier - verify that identifier is for the address we are going to whitelist
+        // for v1 this will do nothing
+        await verifyFVIdentifier(enrollmentIdentifier, gdAddress)
```
Comment (johnsmith-gooddollar) [inline src/server/verification/verificationAPI.js:211]: middleware from example could be also usable here

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/server/verification/verificationAPI.js [hunk 4]
```diff
@@ -37,17 +47,34 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
 
       try {
         const processor = createEnrollmentProcessor(storage, log)
 
-        await verifySignature(enrollmentIdentifier, signature)
-        await processor.enqueueDisposal(user, enrollmentIdentifier, log)
+        // for v2 identifier - verify that identifier is for the address we are going to whitelist
+        await verifyFVIdentifier(enrollmentIdentifier, gdAddress)
+
+        let v2Identifier = enrollmentIdentifier.slice(0, 42)
```
[the remark this answers is not in the document]
Comment (sirpy) [reply src/server/verification/verificationAPI.js:59]: overkill

**Pull request merged**